File: benchmarks/oopsla25/subjects/143.py
import sys

def f143(s: str):
  i = len(s)
  while i > 0:
    if s[i-1] == "a":
      return
    i -= 1
  raise Exception

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f143(my_string)

File: benchmarks/oopsla25/subjects/143.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f143(const char* s) {
    int i = strlen(s);
    while (i > 0) {
        if (s[i - 1] == 'a') {
            return;
        }
        i--;
    }
    exit(1);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f143(my_string);
  return 0;
}
